src/modules/terms/terms.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Inject, Injectable, InternalServerErrorException } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import SharepointConfig from '@ukef/config/sharepoint.config';
+import { RESPONSE } from '@ukef/constants';
+import { GraphService } from '@ukef/modules/graph/graph.service';
+
+type RequiredConfigKeys = 'ukefSharepointName' | 'tfisSiteName' | 'tfisTermStoreId';
+
+@Injectable()
+export class TermsService {
+  constructor(
+    @Inject(SharepointConfig.KEY)
+    private readonly config: Pick<ConfigType<typeof SharepointConfig>, RequiredConfigKeys>,
+    private readonly graphService: GraphService,
+  ) {}
+
+  async postFacilityToTermStore(id: string): Promise<any> {
+    const listItem = {
+      fields: {
+        Title: id,
+      },
+    };
+
+    const { tfisTermStoreId, tfisSiteName, ukefSharepointName } = this.config;
+
+    try {
+      await this.graphService.post<any>({
+        path: `sites/${ukefSharepointName}:/sites/${tfisSiteName}:/lists/${tfisTermStoreId}/items`,
+        listItem,
+      });
+      return { message: RESPONSE.FACILITY_TERM_CREATED };
+    } catch (error: any) {
+      if (error.statusCode === 400) {
+        return { message: RESPONSE.FACILITY_TERM_EXISTS };
```

Comment:
Are we sure that 400 is only thrown when the facility term exists? We might want to check the error message (sometimes graph throws the real error in the `code` value) and then either `return { message: RESPONSE.FACILITY_TERM_EXISTS };` or throw a general GraphUnexpectedException

Reply:
good shout. currently we are returning 200, even when the term already exists in the database. 
I'll add a message on slack to ask the others for thoughts (400 or 200 error)

Reply:
in the meantime, I've changed it so it's using your error handler. + created a new PR since this was merging against the wrong branch